### ISSUE TITLE
Could it be possible to allow for zero blmin?

### ISSUE THREAD

**Issue by annazhukova** [amkozlov/raxml-ng#119]
Dear RAxML-NG developers,

Thank you for a great tool!

For data with low phylogenetic signal and long alignment (e.g. Sars-CoV-2) `--blmin` plays an important role in tree reconstruction as essentially there are many polytomies in the tree (multiple transmissions without mutations), which currently have to be represented by very short branches in order to make the tree binary. I am wondering if it could be possible to allow for `--blmin 0`? Does it pose problem for neighbour tree search? I have also seen that you now output a tree with collapsed short branches. Are those the ones of length `blmin`?

Thanks,

**Comment by amkozlov** [comment]
Dear Anna,

> Thank you for a great tool!

thanks :)

> For data with low phylogenetic signal and long alignment (e.g. Sars-CoV-2) `--blmin` plays an important role in tree reconstruction as essentially there are many polytomies in the tree (multiple transmissions without mutations), which currently have to be represented by very short branches in order to make the tree binary. I am wondering if it could be possible to allow for `--blmin 0`? Does it pose problem for neighbour tree search? 

It is rather a problem for our branch length optimization procedure. However, in practice I guess there would be little difference between true zero (`--blmin 0`) and very small value (eg `--blmin 1e-9` or  `--blmin 1e-12`).  What would make more sense is to collapse short branches into polytomies to reduce search space, but that's a different story. 

> I have also seen that you now output a tree with collapsed short branches. Are those the ones of length `blmin`?

Exactly.

**Comment by annazhukova** [comment]
> It is rather a problem for our branch length optimization procedure. However, in practice I guess there would be little difference between true zero (--blmin 0) and very small value (eg --blmin 1e-9 or --blmin 1e-12). What would make more sense is to collapse short branches into polytomies to reduce search space, but that's a different story.

Indeed, I guess there is not much difference between `--blmin 1e-12` and `--blmin 0`. The advantage of `--blmin 0` would be that it would allow the user to avoid potential problems with a too-high `--blmin` choice (e.g. one corresponding to 1 mutation per alignment) :) So it's more of a potential enhancement. 

>> I have also seen that you now output a tree with collapsed short branches. Are those the ones of length blmin?

> Exactly.

Thanks!
